### PR TITLE
Add notes about dependency updates

### PR DESCRIPTION
**Pull request by uar-dejan-susic**
### Description
During Senior Engineers meeting we decided that this might be needed in future to prevent and mitigate unwanted dependency updates.

### Changes
 - Created new sub paragraph `Updated Dependencies`
  - Added default info value `None`

##### Updated Dependencies
 - None

### Ticket

[XX-1234](https://customink.atlassian.net/browse/XX-1234)

### Optional Tasks

- [ ] Include 🎩 Instructions
- [ ] Update the readme (README.md)
- [ ] Update the API or architecture docs (e.g. docs/api.md)

##### Library-Specific
- [ ] Increment the changelog (CHANGELOG.md)
- [ ] Increment the version number (lib/version.rb)
- [ ] [Release & Tag][release] the version above in Github

[release]: https://docs.github.com/en/github/administering-a-repository/managing-releases-in-a-repository

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -28,6 +28,15 @@ Please include any screenshots that communicate the visual story of the change t
...
 
+### Updated Dependencies
+
+<!--
+Please include any notes that might be helpful for a reviewer to check the dependency changes you might have introduced.
+  - gem version update
+  - new gem introduced
+  - data model update
+-->
+
```
Comment (jfadgen) [inline .github/pull_request_template.md]: Could this be a subsection of `Optional Tasks`, perhaps at the same level of `##### Library-Specific`?

Comment (uar-dejan-susic) [reply .github/pull_request_template.md]: Yes! I was on the fence to put it there or not. Now I am not on the fence any longer! THX

Comment (uar-dejan-susic) [reply .github/pull_request_template.md]: Or perhaps adding one more sub section under the `Notes`? Somehow this is not a `Optional task` thing. This is more of a add more info thing.

Comment (chmurph2) [reply .github/pull_request_template.md]: Yeah, this isn't really task based. Perhaps it's advice that fits under the Changes section?

Comment (uar-dejan-susic) [reply .github/pull_request_template.md]: Moved it under changes and added the default value `None`.

**Review by chmurph2 — APPROVED** [review]

**Comment by uar-dejan-susic** [comment]
Thanks @chmurph2!

I will keep it opened for a bit more so that others can look at it.

**Comment by mpeteuil** [comment]
I wonder if it would be worthwhile to have something about dependency considerations in the guides and then linking in the commented out section here to that? I guess I'm wondering if it's worth being more explicit somewhere (not necessarily here) that there is no free lunch with dependencies, there is always a cost. That might just be me though, others may feel differently.

Either way, nice work!

**Comment by uar-dejan-susic** [comment]
> I wonder if it would be worthwhile to have something about dependency considerations in the guides and then linking in the commented out section here to that? I guess I'm wondering if it's worth being more explicit somewhere (not necessarily here) that there is no free lunch with dependencies, there is always a cost. That might just be me though, others may feel differently.
> 
> Either way, nice work!

Good idea. We could add wiki to this repository, add more info there and link it to template. Would you be interested in adding more info into wiki?

**Review by coding-red-panda — APPROVED** [review]

**Comment by mpeteuil** [comment]
@uar-dejan-susic Yeah, I can do that. I'll try to get something there soon. In the meantime if you'd like to merge this in I'd say feel free and I can update it when that wiki exists. I don't want to hold you up.

**Pull request merged**